rasa/train.py

```diff
@@ -399,15 +477,36 @@ async def _train_core_with_validated_data(
         domain, config = await asyncio.gather(
             file_importer.get_domain(), file_importer.get_config()
         )
+
         async with telemetry.track_model_training(file_importer, model_type="core"):
-            await rasa.core.train(
-                domain_file=domain,
-                training_resource=file_importer,
-                output_path=os.path.join(_train_path, DEFAULT_CORE_SUBDIRECTORY_NAME),
-                policy_config=config,
-                additional_arguments=additional_arguments,
-                interpreter=interpreter,
-            )
+            if training_in_chunks:
+                await rasa.core.train_in_chunks(
+                    domain_file=domain,
+                    training_resource=file_importer,
+                    output_path=Path(_train_path, DEFAULT_CORE_SUBDIRECTORY_NAME),
+                    policy_config=config,
+                    additional_arguments=additional_arguments,
+                    interpreter=interpreter,
+                    number_of_chunks=number_of_chunks,
+                )
+            else:
+                try:
+                    await rasa.core.train(
+                        domain_file=domain,
+                        training_resource=file_importer,
+                        output_path=os.path.join(
+                            _train_path, DEFAULT_CORE_SUBDIRECTORY_NAME
+                        ),
+                        policy_config=config,
+                        additional_arguments=additional_arguments,
+                        interpreter=interpreter,
+                    )
+                except MemoryError:
+                    rasa.shared.utils.io.raise_warning(
+                        "Training failed due to an out of memory error. Consider "
+                        "using the flag '--in-chunks' for training.",
+                        category=UserWarning,
+                    )
```

Comment:
we should still raise the actual exception
```suggestion
                    )
                    raise
```